xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAHD.cpp

```diff
@@ -1029,3 +1057,127 @@ DXGI_COLOR_SPACE_TYPE CProcessorHD::AvToDxgiColorSpace(const DXGIColorSpaceArgs&
 
   return DXGI_COLOR_SPACE_YCBCR_STUDIO_G22_LEFT_P709;
 }
+
+bool CProcessorHD::IsSuperResolutionSuitable(const VideoPicture& picture)
+{
+  if (picture.iWidth > 1920)
+    return false;
+
+  const UINT outputWidth = DX::Windowing()->GetBackBuffer().GetWidth();
+
+  if (outputWidth <= picture.iWidth)
+    return false;
+
+  if (picture.iFlags & DVP_FLAG_INTERLACED)
+    return false;
+
+  if (picture.color_primaries == AVCOL_PRI_BT2020 ||
+      picture.color_transfer == AVCOL_TRC_SMPTE2084 ||
+      picture.color_transfer == AVCOL_TRC_ARIB_STD_B67)
+    return false;
+
+  return true;
+}
+
+void CProcessorHD::TryEnableVideoSuperResolution()
+{
+  if (!m_pVideoContext || !m_pVideoProcessor)
+    return;
+
+  const DXGI_FORMAT format = DX::Windowing()->GetBackBuffer().GetFormat();
+
+  if (format == DXGI_FORMAT_R10G10B10A2_UNORM)
+  {
+    // force 8 bit swap chain temporally as NVIDIA Super Resolution not supports 10 bit
+    DX::DeviceResources::Get()->ApplyDisplaySettings(true);
+    m_forced8bit = true;
+  }
+
```

Comment:
The processor outputs to the intermediate target and the swap chain doesn't need to change.
I know they are coupled currently but don't need to be. This PR is making them more coupled + swap chain creation that could be avoided.

I've been working on something to decouple them (other reasons; HDR to SDR tonemapping with 8 bit output), would you wait for me to finish?
Or if you want to do it yourself (a new virtual function of the renderer to select the intermediate target format per render method dvxa, pixel shaders)

Reply:
Take care with introduce invasive changes in this stage of v21.... now is supposed to be quite stable and PR's like this or AV1 DXVA decoding are not dangerous because they are new features but self contained.

I'm honestly not interested in implementing this nor do I think it's a reason to wait...

Reply:
It wasn't a priority until you posted this change, I'd need a day or two to finish something. It's rather simple and I don't think very risky.

The SDR output of 8 bit is ok, but for an HDR source, the original 10 bits should be preserved when feeding the output shader / tone mapper. That means an intermediate target of 10 bit or float even with an 8 bit back buffer.
It's never going to be a perfect output but reducing the precision before tone mapping likely creates more banding.

Reply:
> but reducing the precision before tone mapping likely creates more banding.

For this reason already exist the setting "Use 10 bit for SDR" because improves video quality and reduce erros due video processing in shaders, etc. If you enable all chain is in 10 bit and is valid for both SDR content or HDR tone mapped to SDR.

Reply:
It's off by default for AMD as workaround for a strange driver problem. Even for Intel/nVidia, it's not intuitive that off can degrade the quality of tone mapping.
Let's move this to the discussion of the PR I'm going to submit